src/Task.js

```diff
@@ -1,46 +1,74 @@
+const fs = require('fs');
+
 const { MigrationError } = require('./error');
 
+// @see: https://loopback.io/doc/en/lb3/Using-database-transactions.html#overview
+const connectorsSupportingTransactions = [
+    'mysql',
+    'postgresql',
+    'mssql',
+    'oracle',
+];
+
 module.exports = class Task {
 
-    constructor(name, version) {
-        this.name = name;
+    get runAlways() {
+        return this.options.runAlways === true;
+    }
+
+    constructor({identifier, version = '1.0.0', options = {} }) {
+        this.identifier = identifier;
         this.version = version;
+        this.options = options;
     }
 
-    async run(dependencies, previous) {
-        const err = new MigrationError('Task:#run() not implemented');
-        return Promise.reject(err);
+    async run(dependencies, previousResult) {
+        throw new MigrationError('Task:#run() not implemented');
     }
 
-    async revert(dependencies) {
-        const err = new MigrationError('Task:#revert() not implemented');
-        return Promise.reject(err);
+    async ensureDependencies(container, dependencyNames = this._getDependencies()) {
```

Comment:
- When is this method called?
- What is the container?

please add documentation